File: docs-examples/codecs-java/src/test/java/io/micronaut/configuration/lettuce/docs/RedisClientCommandsTest.java
package io.micronaut.configuration.lettuce.docs;

import io.lettuce.core.api.StatefulRedisConnection;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
class RedisClientCommandsTest extends AbstractRedisTest {

    @Inject
    RedisClientCommands redisClientCommands;

    @Inject
    StatefulRedisConnection<String, String> connection;

    @Test
    void testCommands() {
        redisClientCommands.commands();

        assertEquals("bar", connection.sync().get("foo"));
    }
}

File: docs-examples/codecs-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/RedisClientCommandsTest.kt
package io.micronaut.configuration.lettuce.docs

import io.lettuce.core.api.StatefulRedisConnection
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautTest
class RedisClientCommandsTest : AbstractRedisTest() {

    @Inject
    lateinit var redisClientCommands: RedisClientCommands

    @Inject
    lateinit var connection: StatefulRedisConnection<String, String>

    @Test
    fun testCommands() {
        redisClientCommands.commands()

        assertEquals("bar", connection.sync().get("foo"))
    }
}
